File: remind_me_be/RemindMe/RemindMe.Domain/Entities/ToDo.cs
namespace RemindMe.Domain.Entities
{
    public class ToDo
    {
        public int Id { get; set; }
        public string? Title { get; set; }
        public string? Description { get; set; }
        public DateTime CreationDate {  get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public bool IsFinished { get; set; }
        public int Difficulty { get; set; }
        public int OwnerId { get; set; }
    }
}

File: remind_me_fe/lib/models/to_do.dart
class ToDo {
  int? id;
  String title;
  String description;
  DateTime creationDate;
  DateTime startDate;
  DateTime? endDate;
  bool isFinished;
  int difficulty;
  int ownerId;

  ToDo({
    this.id,
    this.title = "",
    this.description = "",
    required this.creationDate,
    required this.startDate,
    this.endDate,
    this.isFinished = false,
    required this.difficulty,
    required this.ownerId,
  }) {
    this.creationDate = DateTime.now();
  }

  int? get getId => id;
  String get getTitle => title;
  String get getDescription => description;
  DateTime get getCreationDate => creationDate;
  DateTime get getStartDate => startDate;
  DateTime? get getEndDate => endDate;
  bool get getIsFinished => isFinished;
  int get getDifficulty => difficulty;
  int get getOwnerId => ownerId;
}
